src/meili/geometry_helpers.cc

```diff
@@ -0,0 +1,101 @@
+#include "meili/geometry_helpers.h"
+
+#include <valhalla/midgard/constants.h>
+#include <valhalla/midgard/distanceapproximator.h>
+#include <valhalla/midgard/encoded.h>
+#include <valhalla/midgard/pointll.h>
+#include <valhalla/midgard/util.h>
+
+using namespace valhalla;
+using namespace valhalla::meili;
+using namespace valhalla::midgard;
+
+namespace valhalla {
+namespace meili {
+namespace helpers {
+
+// snapped point, squared distance, segment index, offset
+std::tuple<PointLL, float, typename std::vector<PointLL>::size_type, float>
+Project(const projector_t& p, Shape7Decoder<midgard::PointLL>& shape, float snap_distance) {
```

Comment:
i know you said we'll have to deal with this with double precision too, but actually this method is behind the times. once upon a time everything was single precision floats, but now the main geometric classes are using double precision. specifically PointLL and its various functions are now in `double` including the `Distance` function below. maybe we should update this code as we did with the rest of the code to use double precision so that at least its up to date as well.

Reply:
I think that's a great idea.  I'd love to take that up in a separate "tech debt" ticket.